log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/ThrowablePatternConverter.java

```diff
@@ -235,4 +235,69 @@ protected String getSuffix(final LogEvent event) {
     public ThrowableFormatOptions getOptions() {
         return options;
     }
+
+    private boolean requireAdditionalFormatting(String suffix) {
+        return !options.allLines() || nonStandardLineSeparator || Strings.isNotBlank(suffix) || options.hasPackages();
+    }
+
+    private boolean ignoreElement(final StackTraceElement element, final List<String> ignorePackages) {
+        if (ignorePackages != null) {
+            final String className = element.getClassName();
+            for (final String pkg : ignorePackages) {
+                if (className.startsWith(pkg)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private void appendSuppressedCount(
+            final StringBuilder sb, final int count, final String suffix, final String lineSeparator) {
+        if (count == 1) {
+            sb.append("\t... ");
+        } else {
+            sb.append("\t... suppressed ");
+            sb.append(count);
+            sb.append(" lines");
+        }
+        appendSuffix(sb, suffix);
+        sb.append(lineSeparator);
+    }
+
+    private void appendEntry(
+            final StackTraceElement stackTraceElement,
+            final StringBuilder sb,
+            final String suffix,
+            final String lineSeparator) {
+        sb.append(stackTraceElement.toString());
+        appendSuffix(sb, suffix);
+        sb.append(lineSeparator);
+    }
+
+    private void appendSuffix(StringBuilder buffer, String suffix) {
+        if (Strings.isNotBlank(suffix)) {
+            buffer.append(' ');
+            buffer.append(suffix);
+        }
+    }
+
+    private void truncateLines(StringBuilder sb, String lineSeparator, Integer linesToKeep) {
```

Comment:
Can you implement this as follows, please?

1. Move `ThrowableProxyRenderer.truncateLines(..., TextRenderer)` to `core.util.StringBuilders.truncateLines(..., Consumer<StringBuilder> lineSeparatorInjector)` – This change will cause `bnd-baseline` check to fail. You need to bump the major version in `core/util/package-info.java`.
1. In `ThrowableProxyRenderer`, pass `buffer -> textRenderer.render(lineSeparator, buffer, "Text")` as `lineSeparatorInjector`
1. In `ThrowablePatternConverter`, pass `buffer -> buffer.append(lineSeparator)` as `lineSeparatorInjector`
1. Create `StringBuildersTest` in `log4j-core-test`

Reply:
I can only find `StringBuilders` under `log4j-api/util`, is this the correct one?

Reply:
No, please don't touch to the API. You need to create a new class in `log4j-core`. Thinking about it, better we make it explicit that it is intended for internal usage: Place the class to `org.apache.logging.log4j.core.util.internal` package in `log4j-core`.

Reply:
Got it. Let me create new class in log4j-core 😃